opentracing-spring-web-handler-interceptor/src/main/java/io/opentracing/contrib/spring/web/interceptor/SpanDecorator.java

```diff
@@ -0,0 +1,114 @@
+package io.opentracing.contrib.spring.web.interceptor;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.method.HandlerMethod;
+
+import io.opentracing.Span;
+
+/**
+ * SpanDecorator to decorate span at different stages in filter processing.
+ *
+ * @author Pavol Loffay
+ */
+public interface SpanDecorator {
+
+    /**
+     * This is called in
+     * {@link org.springframework.web.servlet.HandlerInterceptor#preHandle(HttpServletRequest, HttpServletResponse, Object)}.
+     *
+     * @param httpServletRequest request
+     * @param handler handler
+     * @param span current span
+     */
+    void onPreHandle(HttpServletRequest httpServletRequest, Object handler, Span span);
+
+    /**
+     * This is called in
+     * {@link org.springframework.web.servlet.HandlerInterceptor#afterCompletion(HttpServletRequest, HttpServletResponse, Object, Exception)}
+     *
+     * @param httpServletRequest request
+     * @param httpServletResponse response
+     * @param handler handler
+     * @param ex exception
+     * @param span current span
+     */
+    void onAfterCompletion(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, Object handler,
+                           Exception ex, Span span);
+
+    /**
+     * Standard tags used with Web Servlet Tracing Filter
+     */
+    SpanDecorator STANDARD_TAGS = new SpanDecorator() {
+
+        @Override
+        public void onPreHandle(HttpServletRequest httpServletRequest, Object handler, Span span) {
+            Map<String, Object> logs = new HashMap<>(3);
+            logs.put("event", "preHandle");
+            logs.put(MetaDataUtils.HANDLER, handler);
+
+            String metaData = MetaDataUtils.className(handler);
+            if (metaData != null) {
+                logs.put(MetaDataUtils.HANDLER_CLASS_NAME, metaData);
+            }
+
+            metaData = MetaDataUtils.methodName(handler);
+            if (metaData != null) {
+                logs.put(MetaDataUtils.HANDLER_METHOD_NAME, metaData);
+            }
+
+            span.log(logs);
+        }
+
+        @Override
+        public void onAfterCompletion(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
+                                      Object handler, Exception ex, Span span) {
+            Map<String, Object> logs = new HashMap<>(2);
+            logs.put("event", "afterCompletion");
+            logs.put("handler", handler);
```

Comment:
Shouldn't this be same as `onPreHandler` - use `MetaDataUtils.HANDLER`, but also log the class and method names separately?